tracks/python/exercises/darts/mentoring.md

```diff
@@ -0,0 +1,93 @@
+# Mentoring
+
+## Problem and Challenges
+
+The problem asks you to write a function which returns earned points in a single toss of a Darts game.
+The dart's location is given as a Cartesian point (an `x` and `y` value).
+Points are based on the distance from the center.
+
+## Reasonable Solution
+
+```python
+import math
+
+
+def score(x, y):
+    distance = math.sqrt(x * x + y * y)
+    if distance <= 1:
+        return 10
+    if distance <= 5:
+        return 5
+    if distance <= 10:
+        return 1
+    return 0
+```
+
+## Talking Points
+
+### Avoiding the use of `elif`
+
+It can be tempting to use `elif` for the conditionals after the first one.
+However, doing so would be an "unnecessary elif after return", as flagged by Pylint ([R1705][R1705]).
+
+### Minimizing the number of comparisons
+
+The distance checks can compute the upper and lower bound for each point band.
+However, the extra comparions are not needed as they are covered by the prior conditions.
+
+```python
+if 0 <= distance <= 1:
+    return 10
+if 1 < distance <= 5:
+    return 5
+if 5 < distance <= 10:
+    return 1
+```
+
+### Reducing imports
+
+The `math` import is not strictly needed.
+An alternative is to compute `distance = x * x + y * y` and use squared values for the point bands: 1, 25, 100.
+
+```python
+def score(x, y):
+    distance = x * x + y * y
+    if distance <= 1:
+        return 10
+    if distance <= 25:
+        return 5
+    if distance <= 100:
+        return 1
+    return 0
+```
+
+Alternatively, the square root can be computed using `value ** 0.5`.
+
+### Using a loop
+
+The distance checks all have a very similar form: `if distance <= value: return points`.
+Rather than repeating this code three times, a loop can be used.
+
+```python
+import math
+
+
+BANDS = [(1, 10), (5, 5), (10, 1)]
+
+
+def score(x, y):
+    distance = math.sqrt(x * x + y * y)
+    for band, points in BANDS:
+        if distance <= band:
+            return points
+    else:
+        return 0
```

Comment:
Since we have seen the idea "because the linter complains, it is not worth advocating" this code block should be changed for that same reason.

It is a "useless else on loop" as there is no break statement.

Again, the loop is being short circuited, which is a good thing. It is emulating, apparently, what would be a switch or case statement in other languages (where their case statement is not a "waterfall" case statement).  But there is no break that is expected when short-circuiting a loop, from what feedback I am getting from the linter.

Instead, the loop can be responsible for accumulating the score (emulating waterfall style case statement) and we can return the work in its own statement, which makes it clear the loop is responsible for the score, and the return (or any other work needing to be done) is done elsewhere.

And without the reported misuse of `else`.

Reply:
`else` removed.